reporters/dogstatsd.go

```diff
@@ -84,3 +102,23 @@ func NewDogStatsD(config *viper.Viper, logger *logrus.Logger) (*DogStatsD, error
 func NewDogStatsDFromClient(c dogstatsd.Client, r string) *DogStatsD {
 	return &DogStatsD{client: c, region: r}
 }
+
+func (d *DogStatsD) scheduleRestart(timeout time.Duration) {
+	if timeout.Nanoseconds() > 0 {
+		time.AfterFunc(timeout, func() {
+			if err := d.statsdClient.Close(); err != nil {
```

Comment:
I actually was in this doubt too. Looking at the statsd client has a mutex that locks when the data is being written in the UDP connection stream. I could reuse the same mutex. Do you think this is a reliable way to make it safer?

Reply:
But would you be able to reuse that mutex once the client is closed? Otherwise, you could create a new one.

Reply:
You're right! Looking at the internal implementation again, I cannot reuse the same mutex. Even if the mutex is not attached to the connection, it is still being used to lock while the data is streamed.

We still need to create another mutex since the connection can be closed while the data is being flushed.

Reply:
Created sync.Mutex for DogStatsD instance